pkg/sdkserver/sdkserver.go

```diff
@@ -263,6 +267,38 @@ func (s *SDKServer) Run(ctx context.Context) error {
 	return nil
 }
 
+// waitForConnection attempts a GameServer GET every 3s until the client responds.
+// This is a workaround for the informer hanging indefinitely on first LIST due
+// to a flaky network to the Kubernetes service endpoint.
+func (s *SDKServer) waitForConnection(ctx context.Context) error {
+	if s.connected {
+		return nil
+	}
+
+	getWithDeadline := func(ctx context.Context) error {
+		ctx, cancel := context.WithTimeout(ctx, 3*time.Second)
+		defer cancel()
+
+		// Specifically use gameServerGetter since it's the raw client (gameServerLister is the informer).
+		// We use List here to avoid needing permission to Get().
+		_, err := s.gameServerGetter.GameServers(s.namespace).List(ctx, metav1.ListOptions{
+			FieldSelector: fields.OneTermEqualSelector("metadata.name", s.gameServerName).String(),
+		})
+		return err
+	}
+
+	for try := 0; ctx.Err() == nil; try++ {
```

Comment:
(non blocking): This might be nicer as something like https://pkg.go.dev/k8s.io/apimachinery/pkg/util/wait#PollUntilContextTimeout rather than a for loop - since it handles context cancellation etc for you.

Just a thought. Happy to approve otherwise, but figured I'd make the suggestion to see if it resonated.

Reply:
Done. I settled on PollImmediateInfiniteWithContext, which will be deprecated, but isn't yet in the apimachinery version we have (but we also don't have its replacement yet).